x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-ArrayBlock.java.st

```diff
@@ -104,12 +103,28 @@ $endif$
         if (firstValueIndexes == null) {
             return this;
         }
-        int end = firstValueIndexes[getPositionCount()];
-        if (nullsMask == null) {
-            return new $Type$ArrayVector(values, end).asBlock();
+        // TODO use reference counting to share the values
+$if(BytesRef)$
+        final BytesRef scratch = new BytesRef();
+$endif$
+        try (var builder = blockFactory.new$Type$BlockBuilder(firstValueIndexes[getPositionCount()])) {
+            for (int pos = 0; pos < getPositionCount(); pos++) {
+                if (isNull(pos)) {
+                    builder.appendNull();
+                    continue;
+                }
+                int first = getFirstValueIndex(pos);
+                int end = first + getValueCount(pos);
+                for (int i = first; i < end; i++) {
+$if(BytesRef)$
+                    builder.append$Type$(get$Type$(i, scratch));
+$else$
+                    builder.append$Type$(get$Type$(i));
+$endif$
+                }
+            }
+            return builder.mvOrdering(mvOrdering()).build();
```

Comment:
I think we can assign `MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING` for the expanded block?

Reply:
Ah! It shouldn't really matter because it doesn't have duplicated blocks, but that's *technically* correct!